include/net/golioth/lightdb.h

```diff
@@ -20,77 +21,139 @@
+ *
+ * Asynchronously request value from Golioth's LightDB and let @p cb be invoked when such value is
+ * retrieved or some error condtition happens.
+ *
+ * @warning Experimental API
+ *
+ * @param[in] client Client instance
+ * @param[in] path LightDB resource path
+ * @param[in] format Requested format of payload
+ * @param[in] cb Callback executed on response received, timeout or error
+ * @param[in] user_data User data passed to @p cb
+ *
+ * @retval 0 On success
+ * @retval <0 On failure
+ */
+int golioth_lightdb_get_cb(struct golioth_client *client, const uint8_t *path,
+			   enum golioth_content_format format,
+			   golioth_req_cb_t cb, void *user_data);
 
 /**
- * @brief Get value from Golioth's LightDB
+ * @brief Get value from Golioth's LightDB (synchronous, into preallocated buffer)
  *
- * Get value from LightDB and initialize passed CoAP reply handler.
+ * Synchronously get value from Golioth's LightDB and store it into preallocated buffer.
  *
- * @param client Client instance
- * @param path LightDB resource path
- * @param format Requested format of payload
- * @param reply CoAP reply handler object used for notifying about received
- *              value
- * @param reply_cb Reply handler callback
+ * @param[in] client Client instance
+ * @param[in] path LightDB resource path
+ * @param[in] format Requested format of payload
+ * @param[in] data Buffer for received data
+ * @param[in,out] len Size of buffer on input, size of response on output
  *
  * @retval 0 On success
  * @retval <0 On failure
  */
 int golioth_lightdb_get(struct golioth_client *client, const uint8_t *path,
 			enum golioth_content_format format,
-			struct coap_reply *reply, coap_reply_t reply_cb);
+			uint8_t *data, size_t *len);
 
 /**
- * @brief Set value to Golioth's LightDB
+ * @brief Set value to Golioth's LightDB (callback based)
  *
- * Set new value to LightDB.
+ * Asynchronously request to store new value to LightDB and let @p cb be invoked when such value is
+ * actually stored or some error condition happens.
  *
- * @param client Client instance
- * @param path LightDB resource path
- * @param format Format of payload
- * @param data Payload data
- * @param data_len Payload length
+ * @warning Experimental API
+ *
+ * @param[in] client Client instance
+ * @param[in] path LightDB resource path
+ * @param[in] format Requested format of payload
+ * @param[in] data Pointer to data to be set
+ * @param[in] data_len Length of data to be set
+ * @param[in] cb Callback executed on response received, timeout or error
+ * @param[in] user_data User data passed to @p cb
+ *
+ * @retval 0 On success
+ * @retval <0 On failure
+ */
+int golioth_lightdb_set_cb(struct golioth_client *client, const uint8_t *path,
+			   enum golioth_content_format format,
+			   const uint8_t *data, size_t data_len,
+			   golioth_req_cb_t cb, void *user_data);
+
+/**
+ * @brief Set value to Golioth's LightDB (synchronously)
```

Comment:
Are you still planning to add `golioth_lightdb_set_auto` on top of `golioth_lightdb_set`, perhaps in a future PR?

Reply:
I've added #277. I was not working on that last days, it just stayed in a Proof of Concept state. It is something I would like to do in separate PR anyway. I think I can take a little time after this PR gets merged, to cleanup and get it out of "draft" stage.